Merge `dev_test_part13`

LcJuves opened the pull request:
<!-- Thank you for submitting a Pull Request. Please:
* Read our Pull Request guidelines:
  https://github.com/microsoft/vscode/wiki/How-to-Contribute#pull-requests
* Associate an issue with the Pull Request.
* Ensure that the code is up-to-date with the `main` branch.
* Include a description of the proposed changes and how to test them.
-->

KoalaBear84 commented:
Please repo owner, cancel the GitHub workflow as GitHub still didn't took action yet and it runs a cryptominer. And please report the user.

https://docs.github.com/en/actions/managing-workflow-runs/canceling-a-workflow

roblourens committed:
Fix #138427

roblourens committed:
Merge pull request #138428 from microsoft/roblou/fixCellEditorFocus

Always fire onDidChangeState when focusMode is set

mjbvz committed:
Use a unique authority for desktop webview too (#138529)

Fixes #138409

This fix changes webviews on desktop to always use a unique authority instead of reusing the id passed in. In practice, this change should only effect internal webviews, such as those on extension pages or in getting started, since those were the only webviews that were using a hardcoded id

This aligns webviews on desktop with web.

aeschli committed:
make sure theme is loaded when restoring it. Fixes #138422 (#138512)

deepak1556 committed:
revert changes to User agent header (#138556)

Co-authored-by: Sandeep Somavarapu <sasomava@microsoft.com>

Commit:
Try to build snap package

Commit:
Update snapcraft.yaml

jrieken committed:
be more strict when suggesting snippets at word ends, https://github.com/microsoft/vscode/issues/138707

sandy081 committed:
Fix #139013

sandy081 committed:
Fix #139013

sandy081 committed:
Merge pull request #139017 from microsoft/sandy081/fix139013

Pass pre-release info per installed extension to get the latest such extension from marketplace

mjbvz committed:
Pick up TS recovery (#138795)

* Pick up TS recovery

For #138794
s

* Pick up TS 4.5.4

mjbvz committed:
Bump distro and version for 1.63.1 (#139034)

mjbvz committed:
Add yarn file (#139035)

Somehow missed this file with 217f3ec4c2840f958ade9a2c477b850ca2857910

meganrogge committed:
re-enable smoke tests, keep editor fix (#139012) (#139019)

deepak1556 committed:
chore: restore branch for recovery release (#139041)

* Revert "re-enable smoke tests, keep editor fix (#139012) (#139019)"

This reverts commit c4738fedac295235cee339ce669c78c6202edde1.

* Revert "Add yarn file (#139035)"

This reverts commit b3d31564f10eeed9eacf696e8e756d03b8d625dc.

* Revert "Bump distro and version for 1.63.1 (#139034)"

This reverts commit bc22b67787ca6002b1d4a0ba62ff89a851e29250.

* Revert "Pick up TS recovery (#138795)"

This reverts commit 223411ffc738e3f3c9a54ed3693ca047cd5cf749.

* Revert "Merge pull request #139017 from microsoft/sandy081/fix139013"

This reverts commit e7a2ed3cfd4db8f87409785e8631ef8bae994b27, reversing
changes made to 7db1a2b88f7557e0a43fec75b6ba7e50b3e9f77e.

deepak1556 committed:
chore: bump version 1.63.1 (#138976)

alexdima committed:
Go back to using `setImmediate` for background tokenization (#138887)

meganrogge committed:
re-enable smoke tests, keep editor fix (#139012) (#139019) (#139115)

alexdima committed:
Merge pull request #139070 from microsoft/alex/stable-138887

Go back to using `setImmediate` for background tokenization

roblourens committed:
Collapse interactive window cells by default
Fix #139153

sbatten committed:
fixes #138110 (#138796)

TylerLeonhardt committed:
Ensure `forceNewSession` overload gets picked first (#138685) (#138782)

Co-authored-by: Joyce Er <joyce.er@microsoft.com>

mjbvz committed:
Pick up TS 4.5.4 (#139146)

Fixes #138794

roblourens committed:
Merge pull request #139157 from microsoft/roblou/issue139153

Collapse interactive window cells by default

sandy081 committed:
Merge pull request #139176 from microsoft/sandy081/fix139013

Bring back the fix for #139013

jrieken committed:
Merge pull request #138974 from microsoft/joh/fix/138707

be more strict when suggesting snippets at word ends

deepak1556 committed:
chore: bump version to 1.63.2 (#139179)

Commit:
Merge branch 'fetch' into dev/liangchengj

Commit:
Update hash

Commit:
Comment set `darwinExecutable`

Commit:
Comment set `darwinBundleIdentifier`

Commit:
Comment set `serverGreeting`

Commit:
Pre-release `dev_test_part12`

Commit:
Use `extract-zip` command instead of `unzip` command

Commit:
Update the method of obtaining `VS_ZIP`

Commit:
Fix errors

Commit:
Try to fix the error

Commit:
Replaced by `wget`

Commit:
Fix errors

Commit:
Pre-release `dev_test_part13`

Commit:
Skip `linux-ia32`

Commit:
macOS uses `sevenzip` instead of `zip` compression products

Commit:
Windows platforms use `bash` builds instead

Commit:
Optimize build steps

Commit:
Remove error and unused step

Commit:
Optimize build steps

Commit:
Fix errors

Commit:
Fix an error
